claat/parser/trim.go

```diff
@@ -174,6 +174,17 @@ func splitSpaceRight(s string) (v string, sp string) {
 	return string(rs), ""
 }
 
+func requiresSpacer(a, b types.Node) bool {
+	t1, ok1 := a.(*types.TextNode)
+	t2, ok2 := b.(*types.TextNode)
+
+	if !(ok1 && ok2) {
+		return false
+	}
+
+	return (t1.Bold && t1.Bold == t2.Bold) || (t1.Italic && t1.Italic == t2.Italic)
```

Comment:
If I read right, this is asking "are `t1` and `t2` either both bold or both italic?" I think it would be more readable to test those directly instead of using `==`.

Reply:
Yeah this is way better - fixed.